Projects/views.py

```diff
@@ -12,16 +12,47 @@
 from .models import Issues,Prs
 
 def home(request):
-	issues = Issues.objects.all().order_by('points')
-	paginator = Paginator(issues, 15) # Show 15 issues per page
-	page = request.GET.get('page', 1)
-	try:
-		issues = paginator.get_page(page)
-	except PageNotAnInteger:
-		issues = paginator.get_page(1)
-	except EmptyPage:
-		issues = paginator.get_page(paginator.num_pages)
-	return render(request, 'Projects/home.html', {'issues':issues})
+	if request.method == 'POST':
+		# Getting value of the clicked option
+		val = request.POST['value']
+
+		print(val)
+		# First it checks for project name
+		issues = Issues.objects.filter(title_project=val)
+		# Then it goes for mentor name
+		if (issues.count() == 0):
+			try:
+				mentor = User.objects.get(username=val)
+				issues = Issues.objects.filter(mentor=mentor)
+			# else ordered point wise
+			except User.DoesNotExist:
+				issues = Issues.objects.all().order_by('points')
+
+		paginator = Paginator(issues, 15)  # Show 15 issues per page
+		page = request.GET.get('page', 1)
+		try:
```

Comment:
Too much redendent code :smile: try not to have same line of codes twice

Reply:
Line no 31-39 and 46-54 have same lines of code

Reply:
But both are for two different response.......the lines(46-54) are required when it is not POST request.
Without this how could possibly the home page will fetch issues. 

And everything it apart its working without any bug

Reply:
Though GUI looks good but are you sure this is working ? 

It won't work :smile: 
Some bugs - 
1. For infinite scroll a get request is sent for fetching next issues, these new issues won't follow the filter properties as the request sent for fetching new issues on infinite scroll doesn't know what to filter? 
2. How will you filter all issues by "xyz" project of "abcd" mentor i.e one can filter only on the basis of one property.

Some suggestions - 
1. Try to have checkboxes, filter on the basis of all checked/unchecked properties
2. For infinite scroll to work you need to send all the checked attributes with page number, send the filter properties names in dictionary and use them while fetching new issues on infinite scroll
3. Use GET method, no need to use POST

You'll surely get partial points. For this PR to be merged fix the bugs :smiley: